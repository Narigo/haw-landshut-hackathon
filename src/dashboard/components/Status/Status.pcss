.status {
  color: white;
  padding: 0.5em 1em;

  & p {
    display: inline-block;
  }
}
