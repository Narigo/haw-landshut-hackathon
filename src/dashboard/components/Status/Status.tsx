import * as React from "react";

import PresentationComponent, {PresentationComponentProps} from "../PresentationComponent/PresentationComponent";

import * as statusStyles from "./Status.pcss";

interface StatusProps extends PresentationComponentProps {
  statusText: string;
  status: string;
}

export default class Status extends PresentationComponent<StatusProps, {}> {
  public render() {
    const {statusText, children, status} = this.props;

    const statusHexcodes = {
      ok: "#55b00f",
      error: "#df0a00",
      warning: "#df8f00"
    };

    const statusInlineStyles = {
      background: statusHexcodes[status]
    };

    return (
      <div style={statusInlineStyles} className={statusStyles.status}>
        <div className={statusStyles.text}>{statusText}</div>
        {children}
      </div>
    );
  }
}
