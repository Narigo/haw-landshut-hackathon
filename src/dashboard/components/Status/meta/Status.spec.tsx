import * as React from "react";
import * as renderer from "react-test-renderer";

import Status from "../Status";

describe("Status", () => {
  test("will render", () => {
    const component = renderer.create(<Status />);

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
