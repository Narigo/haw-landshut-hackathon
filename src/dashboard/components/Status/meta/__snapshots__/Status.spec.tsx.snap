// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status will render 1`] = `
<div>
  test
</div>
`;
